### ISSUE TITLE
the leastauthority.com repo must be restructured to reflect excision of blog source

### ISSUE THREAD

**Issue by zancas** [LeastAuthority/leastauthority.com#263]
zancas commented 6 days ago

Actually this is already accomplished on branch: 238_remove_the_blog_2 so simply deploying that branch closes this issue.
zancas
zancas commented 2 days ago

I propose that I:

(1) check out current production master on "testing"
(2) run the servers (flapp and web)
(3) view the blog
(4) checkout 238_remove_the_blog_2
(5) re View the site
(6) run update_testing_blog.py
(7) re View the site
(8) ask for a pair-peer
(9) clone blog_source to production
(10) checkout 238_remove_the_blog_2 on production CAUSING TEMPORARY OUTAGE
(11) run update_prodution_blog.py (fixing the outage)
zancas
zancas commented 2 days ago

I'll start with steps 1-7, now.
zancas
zancas commented 2 days ago

Ammendment:

I set the leastauthority.com repo to a commit which has had its blog_source directory removed (with git rm), I pushed that repo to testing and checked out the commit that did not have the blog_source directory.

The blog_source directory which was in the working directory prior to the checkout was unaffected. Therefore this process will not cause an outage.
zancas
zancas commented 2 days ago

Even without the --delete-output-directory flag pelican removes the target output directory.
zancas
zancas commented 2 days ago

Instead of git checkout 238_... on production we'll merge 238_... into master on github.

We'll git push --verbose production master:master maintaining synchronization.

We'll:
(1) ssh into production
(2) git merge --ff-only master on production
zancas
zancas commented 2 days ago

A clue from /var/log/auth.log on rho:

Jul  8 03:19:43 ip-172-31-47-120 sshd[20735]: error: Could not load host key: /etc/ssh/ssh_host_ed25519_key
Jul  8 03:19:43 ip-172-31-47-120 sshd[20735]: Connection closed by 67.176.52.224 [preauth]

zancas zancas referenced this issue in LeastAuthority/leastauthority.com a day ago
Open
openssh pubkey ECDSA vs RSA mismatch #10
zancas
zancas commented a day ago

OK, I've figured out the SSH authentication issue. (I THINK).

If I:

mv ~/.ssh/known_hosts $BACKUP

and run one of the update_.._blog.py tools, I get the same error that blocked us earlier in the deployment.

If I then ssh website@testing.leastauthority.com, I get:

 0 git_branch: master
/home/arc/.ssh :
$ ssh website@testing.leastauthority.com
The authenticity of host 'testing.leastauthority.com (107.21.225.70)' can't be established.
ECDSA key fingerprint is c8:36:84:9f:20:be:3d:71:c9:e4:15:a9:76:4e:91:17.
Are you sure you want to continue connecting (yes/no)? yes
Warning: Permanently added 'testing.leastauthority.com,107.21.225.70' (ECDSA) to the list of known hosts.

Notice that the key used is an ECDSA key.

If I then run:

python ./update_testing_blog.py

I get the same error as I got when there was no known_hosts file.

Now mv known_hosts aside again and run:

 130 git_branch: master
/home/arc/.ssh :
$ ssh-keyscan -t rsa testing.leastauthority.com >> ~/.ssh/known_hosts
# testing.leastauthority.com SSH-2.0-OpenSSH_6.6p1 Ubuntu-2ubuntu1

 0 git_branch: master
/home/arc/.ssh :
$ cat ~/.ssh/known_hosts
testing.leastauthority.com ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQCWvLycjtQeZJpXbhSm/Y5fzbe3JV36CKehVzJQvqrJYg9oOpCwneXFIrIEhOS5Lg6OwXxe1jCqAYL6r+UOjTvMyGc971FU1d1qID71LcTdAiiI2k+8KeJjuOdH2P8WRIL4XQ21xQbqx/o7kb3OTeWDpBdzqrX4bSRwZ5MyzwccUi8ff9kNGjfQk97dt/e+PfxbjOpMm9JCFWGvcXJLvKo8gQCm1q7oajUzexCo8tuwu0TFcOlXCFUore5j8d/eUo0yvJZwgQuoyoSpok/o5sIRoAdVTM3eufkcy1DM0vFDJKP+OGbt+TLBEQo5ibOWEQTb1jpsPcxLVhcFnEIIjFKV

I still get the same error, which I am confused about, perhaps it's because the keyscan result was matched to a domain name... /me checks this hypothesis....

Yes, the "mv known_hosts" test confirms this, so the first argument to "update_blog" must be matched to a known host (either IPv4 or domain-name).

It seems that for multihoming, and switching servers, we should map to the domain name for infrastructure servers... indeed SSEC2s have publicly routable domain-names as well IIRC, so I'll make a new issue.

Once the public key in "known_hosts" is of type rsa, and once it's mapped to the IPv4 address passed to the function update_blog

the script authenticates as expected.
zancas
zancas commented a day ago

I don't know which software is unable to check ecdsa pubkeys.

paramiko? fabric?
zancas
zancas commented 23 hours ago

Next problem, not yet understood:

website@ip-172-31-47-120:~/blog_source$ python ./render_blog.py 
WARNING: LOCALE option doesn't contain a correct value
ERROR: Skipping /home/website/blog_source/README.md: could not find information about 'title'
-> copying /home/website/blog_source/themes/lae/static to /home/website/leastauthority.com/content/blog/theme
-> writing /home/website/leastauthority.com/content/blog/feeds/all.atom.xml
-> writing /home/website/leastauthority.com/content/blog/feeds/essay.atom.xml
-> writing /home/website/leastauthority.com/content/blog/feeds/events.atom.xml
-> writing /home/website/leastauthority.com/content/blog/feeds/letters.atom.xml
-> writing /home/website/leastauthority.com/content/blog/feeds/news.atom.xml
-> writing /home/website/leastauthority.com/content/blog/feeds/press.atom.xml
-> writing /home/website/leastauthority.com/content/blog/feeds/press-releases.atom.xml
-> writing /home/website/leastauthority.com/content/blog/feeds/all-en.atom.xml
-> writing /home/website/leastauthority.com/content/blog/least_authority_releases_s4+_as_part_of_reset_the_net.html
-> writing /home/website/leastauthority.com/content/blog/least_authority_performs_security_audit_for_globaleaks.html
-> writing /home/website/leastauthority.com/content/blog/least_authority_performs_security_audit_for_cryptocat.html
-> writing /home/website/leastauthority.com/content/blog/BLAKE2-harder-better-faster-stronger-than-MD5.html
-> writing /home/website/leastauthority.com/content/blog/least_authority_performs_security_audit_for_spideroak.html
-> writing /home/website/leastauthority.com/content/blog/lafs_summit_san_francisco_nov_11_13.html
-> writing /home/website/leastauthority.com/content/blog/lafs_featured_eff_tech_blog.html
-> writing /home/website/leastauthority.com/content/blog/least_authority_joins_open_invention_network.html
-> writing /home/website/leastauthority.com/content/blog/s4_press_on_linuxbsdos.html
-> writing /home/website/leastauthority.com/content/blog/open_letter_silent_circle.html
-> writing /home/website/leastauthority.com/content/blog/least_authority_press_motherboard_vice_com.html
-> writing /home/website/leastauthority.com/content/blog/lafs_press_computerworld.html
-> writing /home/website/leastauthority.com/content/blog/least-authority-announces-prism-proof-storage-service.html
CRITICAL: 'articles_next_page' is undefined
Traceback (most recent call last):
  File "./render_blog.py", line 14, in <module>
    render_blog()
  File "./render_blog.py", line 10, in render_blog
    subprocess.check_call(['pelican', '-v', '-s', 'conf.py'])
  File "/usr/lib/python2.7/subprocess.py", line 540, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['pelican', '-v', '-s', 'conf.py']' returned non-zero exit status 1
website@ip-172-31-47-120:~/blog_source$ 

zancas
zancas commented 22 hours ago

By testing on 'testing' I confirm that pelican 3.2.2 has this issue and 3.3.0 does not.
zancas
zancas commented 22 hours ago

Hmmm, here's what testing says about pelican post pip install pelican==3.3.0
## website@ip-10-142-174-67:~/blog_source$ pip show pelican

Name: pelican
Version: 3.3
Location: /usr/local/lib/python2.7/dist-packages
Requires: jinja2, blinker, pygments, docutils, python-dateutil, pytz, six, feedgenerator, unidecode
website@ip-10-142-174-67:~/blog_source$ pelican --version  
3.4.0
website@ip-10-142-174-67:~/blog_source$ 

nathan-at-least
nathan-at-least commented 22 hours ago

What does which pelican say?
zancas
zancas commented 22 hours ago

website@ip-10-142-174-67:~$ which pelican
/usr/local/bin/pelican

nathan-at-least
nathan-at-least commented 22 hours ago

I'm going to install pelican into a fresh virtualenv to see if my local system shows the same version issue.
zancas
zancas commented 22 hours ago

I ran:

pip uninstall pelican
which pelican

on testing and there was no result.

Then I ran:

pip install pelican==3.3.0

and then:

root@ip-10-142-174-67:/home/ubuntu# which pelican
/usr/local/bin/pelican
root@ip-10-142-174-67:/home/ubuntu# pelican --version
3.4.0
## root@ip-10-142-174-67:/home/ubuntu# pip show pelican

Name: pelican
Version: 3.3
Location: /usr/local/lib/python2.7/dist-packages
Requires: jinja2, blinker, pygments, docutils, python-dateutil, pytz, six, feedgenerator, unidecode
root@ip-10-142-174-67:/home/ubuntu# 

nathan-at-least
nathan-at-least commented 22 hours ago

If I do a fresh pip install I do not see this issue:

$ mkdir tmpvenv

$ virtualenv !$
virtualenv tmpvenv
Using real prefix '/usr'
New python executable in tmpvenv/bin/python
Installing Setuptools..............................................................................................................................................................................................................................done.
Installing Pip.....................................................................................................................................................................................................................................................................................................................................done.

$ . ./tmpvenv/bin/activate
(tmpvenv)
$ which pip
/home/n/tmp/tmpvenv/bin/pip
(tmpvenv)
$ which pelican
(tmpvenv)
$ pip install 'pelican==3.3.0'
Downloading/unpacking pelican==3.3.0
  Downloading pelican-3.3.tar.gz (195kB): 195kB downloaded
  Storing download in cache at /home/n/.pip/cache/https%3A%2F%2Fpypi.python.org%2Fpackages%2Fsource%2Fp%2Fpelican%2Fpelican-3.3.tar.gz
  Running setup.py egg_info for package pelican

Downloading/unpacking feedgenerator>=1.6 (from pelican==3.3.0)
  Downloading feedgenerator-1.7.tar.gz
  Storing download in cache at /home/n/.pip/cache/https%3A%2F%2Fpypi.python.org%2Fpackages%2Fsource%2Ff%2Ffeedgenerator%2Ffeedgenerator-1.7.tar.gz
  Running setup.py egg_info for package feedgenerator

```
warning: no files found matching '*' under directory 'tests'
warning: no previously-included files matching '*.pyc' found under directory 'tests'
```

Downloading/unpacking jinja2>=2.7 (from pelican==3.3.0)
  Downloading Jinja2-2.7.3.tar.gz (378kB): 378kB downloaded
  Storing download in cache at /home/n/.pip/cache/https%3A%2F%2Fpypi.python.org%2Fpackages%2Fsource%2FJ%2FJinja2%2FJinja2-2.7.3.tar.gz
  Running setup.py egg_info for package jinja2

```
warning: no files found matching '*' under directory 'custom_fixers'
warning: no previously-included files matching '*' found under directory 'docs/_build'
warning: no previously-included files matching '*.pyc' found under directory 'jinja2'
warning: no previously-included files matching '*.pyc' found under directory 'docs'
warning: no previously-included files matching '*.pyo' found under directory 'jinja2'
warning: no previously-included files matching '*.pyo' found under directory 'docs'
```

Downloading/unpacking pygments (from pelican==3.3.0)
  Downloading Pygments-1.6.tar.gz (1.4MB): 1.4MB downloaded
  Storing download in cache at /home/n/.pip/cache/https%3A%2F%2Fpypi.python.org%2Fpackages%2Fsource%2FP%2FPygments%2FPygments-1.6.tar.gz
  Running setup.py egg_info for package pygments

Downloading/unpacking docutils (from pelican==3.3.0)
  Downloading docutils-0.11.tar.gz (1.6MB): 1.6MB downloaded
  Storing download in cache at /home/n/.pip/cache/https%3A%2F%2Fpypi.python.org%2Fpackages%2Fsource%2Fd%2Fdocutils%2Fdocutils-0.11.tar.gz
  Running setup.py egg_info for package docutils

```
warning: no files found matching 'MANIFEST'
warning: no files found matching '*' under directory 'extras'
warning: no previously-included files matching '.cvsignore' found under directory '*'
warning: no previously-included files matching '*.pyc' found under directory '*'
warning: no previously-included files matching '*~' found under directory '*'
warning: no previously-included files matching '.DS_Store' found under directory '*'
```

Downloading/unpacking pytz>=0a (from pelican==3.3.0)
  Downloading pytz-2014.4.tar.bz2 (159kB): 159kB downloaded
  Storing download in cache at /home/n/.pip/cache/https%3A%2F%2Fpypi.python.org%2Fpackages%2Fsource%2Fp%2Fpytz%2Fpytz-2014.4.tar.bz2
  Running setup.py egg_info for package pytz

Downloading/unpacking blinker (from pelican==3.3.0)
  Downloading blinker-1.3.tar.gz (91kB): 91kB downloaded
  Storing download in cache at /home/n/.pip/cache/https%3A%2F%2Fpypi.python.org%2Fpackages%2Fsource%2Fb%2Fblinker%2Fblinker-1.3.tar.gz
  Running setup.py egg_info for package blinker

Downloading/unpacking unidecode (from pelican==3.3.0)
  Downloading Unidecode-0.04.16.tar.gz (200kB): 200kB downloaded
  Storing download in cache at /home/n/.pip/cache/https%3A%2F%2Fpypi.python.org%2Fpackages%2Fsource%2FU%2FUnidecode%2FUnidecode-0.04.16.tar.gz
  Running setup.py egg_info for package unidecode

Downloading/unpacking six (from pelican==3.3.0)
  Using download cache from /home/n/.pip/cache/https%3A%2F%2Fpypi.python.org%2Fpackages%2Fsource%2Fs%2Fsix%2Fsix-1.7.3.tar.gz
  Running setup.py egg_info for package six

```
no previously-included directories found matching 'documentation/_build'
```

Downloading/unpacking markupsafe (from jinja2>=2.7->pelican==3.3.0)
  Downloading MarkupSafe-0.23.tar.gz
  Storing download in cache at /home/n/.pip/cache/https%3A%2F%2Fpypi.python.org%2Fpackages%2Fsource%2FM%2FMarkupSafe%2FMarkupSafe-0.23.tar.gz
  Running setup.py egg_info for package markupsafe

Installing collected packages: pelican, feedgenerator, jinja2, pygments, docutils, pytz, blinker, unidecode, six, markupsafe
  Running setup.py install for pelican

```
Installing pelican script to /home/n/tmp/tmpvenv/bin
Installing pelican-import script to /home/n/tmp/tmpvenv/bin
Installing pelican-quickstart script to /home/n/tmp/tmpvenv/bin
Installing pelican-themes script to /home/n/tmp/tmpvenv/bin
```

  Running setup.py install for feedgenerator

```
warning: no files found matching '*' under directory 'tests'
warning: no previously-included files matching '*.pyc' found under directory 'tests'
```

  Running setup.py install for jinja2

```
warning: no files found matching '*' under directory 'custom_fixers'
warning: no previously-included files matching '*' found under directory 'docs/_build'
warning: no previously-included files matching '*.pyc' found under directory 'jinja2'
warning: no previously-included files matching '*.pyc' found under directory 'docs'
warning: no previously-included files matching '*.pyo' found under directory 'jinja2'
warning: no previously-included files matching '*.pyo' found under directory 'docs'
```

  Running setup.py install for pygments

```
Installing pygmentize script to /home/n/tmp/tmpvenv/bin
```

  Running setup.py install for docutils
    changing mode of build/scripts-2.7/rst2html.py from 644 to 755
    changing mode of build/scripts-2.7/rst2s5.py from 644 to 755
    changing mode of build/scripts-2.7/rst2latex.py from 644 to 755
    changing mode of build/scripts-2.7/rst2xetex.py from 644 to 755
    changing mode of build/scripts-2.7/rst2man.py from 644 to 755
    changing mode of build/scripts-2.7/rst2xml.py from 644 to 755
    changing mode of build/scripts-2.7/rst2pseudoxml.py from 644 to 755
    changing mode of build/scripts-2.7/rstpep2html.py from 644 to 755
    changing mode of build/scripts-2.7/rst2odt.py from 644 to 755
    changing mode of build/scripts-2.7/rst2odt_prepstyles.py from 644 to 755

```
warning: no files found matching 'MANIFEST'
warning: no files found matching '*' under directory 'extras'
warning: no previously-included files matching '.cvsignore' found under directory '*'
warning: no previously-included files matching '*.pyc' found under directory '*'
warning: no previously-included files matching '*~' found under directory '*'
warning: no previously-included files matching '.DS_Store' found under directory '*'
changing mode of /home/n/tmp/tmpvenv/bin/rst2html.py to 755
changing mode of /home/n/tmp/tmpvenv/bin/rst2man.py to 755
changing mode of /home/n/tmp/tmpvenv/bin/rst2pseudoxml.py to 755
changing mode of /home/n/tmp/tmpvenv/bin/rst2odt.py to 755
changing mode of /home/n/tmp/tmpvenv/bin/rstpep2html.py to 755
changing mode of /home/n/tmp/tmpvenv/bin/rst2xml.py to 755
changing mode of /home/n/tmp/tmpvenv/bin/rst2xetex.py to 755
changing mode of /home/n/tmp/tmpvenv/bin/rst2latex.py to 755
changing mode of /home/n/tmp/tmpvenv/bin/rst2odt_prepstyles.py to 755
changing mode of /home/n/tmp/tmpvenv/bin/rst2s5.py to 755
```

  Running setup.py install for pytz

  Running setup.py install for blinker

  Running setup.py install for unidecode

  Running setup.py install for six

```
no previously-included directories found matching 'documentation/_build'
```

  Running setup.py install for markupsafe

```
building 'markupsafe._speedups' extension
x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -I/usr/include/python2.7 -c markupsafe/_speedups.c -o build/temp.linux-x86_64-2.7/markupsafe/_speedups.o
x86_64-linux-gnu-gcc -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-z,relro -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -D_FORTIFY_SOURCE=2 -g -fstack-protector --param=ssp-buffer-size=4 -Wformat -Werror=format-security build/temp.linux-x86_64-2.7/markupsafe/_speedups.o -o build/lib.linux-x86_64-2.7/markupsafe/_speedups.so
```

Successfully installed pelican feedgenerator jinja2 pygments docutils pytz blinker unidecode six markupsafe
Cleaning up...
(tmpvenv)
$ which pelican
/home/n/tmp/tmpvenv/bin/pelican
(tmpvenv)
$ pelican --version
3.3.0
(tmpvenv)

nathan-at-least
nathan-at-least commented 21 hours ago
## $ pip show pelican

Name: pelican
Version: 3.3
Location: /home/n/tmp/tmpvenv/lib/python2.7/site-packages
Requires: feedgenerator, jinja2, pygments, docutils, pytz, blinker, unidecode, six
(tmpvenv)

nathan-at-least
nathan-at-least commented 21 hours ago

What happens if you tell bash to hash -r? Could it possible be caching a reference to a different script?
zancas
zancas commented 21 hours ago

root@ip-10-142-174-67:/home/ubuntu# pelican --version  
3.4.0

nathan-at-least
nathan-at-least commented 21 hours ago

Za linked to https://stackoverflow.com/questions/5226311/installing-specific-package-versions-with-pip

Does pip uninstall pelican && pip install 'pelican==3.3.0' help?
zancas
zancas commented 21 hours ago

The -I flag seems to make pip install -I ignore previously met dependencies.

But I still have the same result.
zancas
zancas commented 21 hours ago

The above invocation with the "'" 's doesn't help.
zancas
zancas commented 21 hours ago

How do I find out the source URL of a pip package?
Maybe, as with that stackoverflow ticket... hmmm... no that doesn't make sense.
zancas
zancas commented 21 hours ago

After:

root@ip-10-142-174-67:/home/ubuntu# pip uninstall pelican && pip install -I 'pelican==3.2.2'

root@ip-10-142-174-67:/home/ubuntu# pelican --version  
3.2.2
## root@ip-10-142-174-67:/home/ubuntu# pip show pelican                                        

Name: pelican
Version: 3.2.2
Location: /usr/local/lib/python2.7/dist-packages
Requires: feedgenerator, jinja2, pygments, docutils, pytz, blinker, unidecode, six

So it's specific to 3.3.0!!

BTW 3.4.0 is "LATEST".
zancas
zancas commented 21 hours ago

And, also specific to the testing machine (so far).
zancas
zancas commented 21 hours ago

Where does one report pelican/pip bugs....
zancas
zancas commented 21 hours ago

Hmmm... this link is live:

https://pypi.python.org/pypi/pelican/3.2.2
nathan-at-least
nathan-at-least commented 21 hours ago

Why does it work on my machine?

$ pip --version
pip 1.4.1 from /home/n/tmp/tmpvenv/lib/python2.7/site-packages (python 2.7)
(tmpvenv)

zancas
zancas commented 21 hours ago

Ha! Jynx:

root@ip-10-142-174-67:/home/ubuntu# pip --version
pip 1.5.4 from /usr/lib/python2.7/dist-packages (python 2.7)

zancas
zancas commented 21 hours ago

(LeastAuthority_env) 0 git_branch: master
/home/arc/LeastAuthority_env/LeastAuthority/website/leastauthority.com :
$ pip --version
pip 1.3.1 from /home/arc/LeastAuthority_env/lib/python2.7/site-packages/pip-1.3.1-py2.7.egg (python 2.7)

zancas
zancas commented 21 hours ago

I'll try changing the version of pip on testing...
zancas
zancas commented 21 hours ago

SIgh... how to install a particular apt package version....
zancas
zancas commented 21 hours ago

http://debian-handbook.info/browse/wheezy/sect.apt-get.html
zancas
zancas commented 21 hours ago

This link is not available:

https://pypi.python.org/pypi/pelican/3.3.0

So how did my local pip install 3.3.0 ?

/me runs:

(LeastAuthority_env) 0 git_branch: master
/home/arc/LeastAuthority_env/LeastAuthority/website/leastauthority.com :
$ pip install -v -I pelican==3.3.0 | less

zancas
zancas commented 21 hours ago

Phewf even 1 "v" is pretty verbose!
zancas
zancas commented 20 hours ago

I can't seem to replicate this elsewhere, but I've replicated it many times with:

pip 1.5.4

On testing.
zancas
zancas commented 20 hours ago

Sigh... well pelican 3.4.0 renders successfully when used by update_testing_blog.py
zancas
zancas commented 19 hours ago

I installed pelican 3.4.0 on rho like this:

root@ip-172-31-47-120:/home/ubuntu# pip uninstall pelican
root@ip-172-31-47-120:/home/ubuntu# pip install -I pelican==3.4.0

then I ran:

python update_production_blog.py
zancas
zancas commented 33 minutes ago

Nej, I think you should close this issue.

**Comment by Liz315** [comment]
Will be overcome by the new site.
